edgexfoundry/device-modbus-go
# fix(snap): update snap v2 support

This update fixes a few issues for the v2 update. Specifically:

- set adaptor:full to resolve issue with libzmq library not loading
- add slot for SecretStore/Vault token and set matching environment variable
- use the git tag for versioning, rather than the VERSION file
- set release name to Ireland in pre-refresh hook
- add support for res/devices and res/profiles directories in install hook and snapcraft.yaml
- bump epoch to 2

Test by doing:

1. Run against the draft PR version of edgex-go from https://github.com/edgexfoundry/edgex-go/pull/3525

Before building that snap, update `snapcraft.yaml`, adding an environment variable to create a token for device-modbus:

```
  security-secretstore-setup:
  ...
    environment:
      ...
      ADD_SECRETSTORE_TOKENS: "device-modbus"
```

2. build and install the snap

```
sudo snap install ./edgex-device-modbus.3.2-dev.18_amd64.snap --dangerous
```

3. As this is a side-loaded snap, we need to connect the interface manually for the token

```
snap connect edgexfoundry:edgex-secretstore-token edgex-device-modbus:edgex-secretstore-token 
```

4. and for the same reason, we also need to regenerate the tokens to get a new device-modbus token 

```
sudo snap restart edgexfoundry.security-secretstore-setup
```

5. Verify that you have a token

```
sudo snap run --shell edgex-device-modbus.device-modbus
ls $SNAP_DATA/device-modbus/
exit
```

6. restart the service and confirm it starts

```
sudo snap restart edgex-device-modbus
```

Note that restarting secret store is not supported generally, so doing that will break consul etc.

Signed-off-by: Siggi Skulason <siggi.skulason@canonical.com>

## PR Checklist
Please check if your PR fulfills the following requirements:


**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-modbus-go/blob/master/.github/Contributing.md

## What is the current behavior?



## Issue Number:


## What is the new behavior?


## Does this PR introduce a breaking change?



## New Imports



## Specific Instructions


## Other information
